src/power_grid_model/validation/utils.py

```diff
@@ -243,7 +246,7 @@ def get_valid_ids(data: SingleDataset, ref_components: Union[str, List[str]]) ->
     """
     # For convenience, ref_component may be a string and we'll convert it to a 'list' containing that string as it's
     # single element.
-    if isinstance(ref_components, str):
+    if not isinstance(ref_components, list):
```

Comment:
While it is correct, there is no need to modify logic :D
```suggestion
    if isinstance(ref_components, (str, ComponentTypes)):
```

Reply:
I modified it because `isinstance()` gives me trouble with `ComponentTypes` :/. Any other suggestion to keep the logic is welcomed.

Reply:
what kind of trouble, specifically?

Reply:
@mgovers @nitbharambe Neither mypy nor `isinstance()`  like the fact that `ComponentTypes = Union[ComponentType, ComponentTypeLiterals]`. However, with the new definition of `class ComponentType(str, Enum)` this is fine. Nitish suggestion has been implemented.

Reply:
@mgovers  @nitbharambe  Neither mypy nor isinstance() like the fact that ComponentTypes = Union[ComponentType, ComponentTypeLiterals]. However, with the new definition of `class ComponentType(str, Enum)` this is fine. Nitish suggestion has been implemented.